doc/source/user-environment.rst

```diff
@@ -219,27 +221,54 @@ variables.
 Pre-populating user's ``$HOME`` directory with notebooks
 --------------------------------------------------------
 
-By default, the contents of ``$HOME`` in the docker image are hidden by
-the contents of the per-user persistent volume. If you want to, you can
-execute a command before the notebook starts each time and copy the files
-you want from your image to the user's home directory.
+If persistent storage is enabled (on by default) then the contents of
+``$HOME`` in the docker image will be hidden to the user. In order to
+pre-populate the user's filesystem, you can include commands
+to be run each time a user starts their server. To do so, use the following
+pattern in ``config.yaml``:
 
-If you were using the repo2docker method of building an image and wanted
-your git repo copied on first use to the user's home directory, you can
-use the following in your ``config.yaml`` file:
+.. code-block:: bash
 
-   .. code-block:: bash
+   singleuser:
+     lifecycleHooks:
+       postStart:
+         exec:
+           command: ["your", "command", "here"]
 
-      singleuser:
-        lifecycleHooks:
-          postStart:
-            exec:
-              command: ["/bin/sh", "-c", "test -f $HOME/.copied || cp -Rf /srv/app/src/. $HOME/; touch $HOME/.copied"]
+We recommend using the tool `nbgitpuller <https://github.com/data-8/nbgitpuller>`_
+to synchronize a folder in your user's filesystem with a ``git`` repository.
+
+To use ``nbgitpuller``, first make sure that you `install it in your Docker
+image <https://github.com/data-8/nbgitpuller#installation>`_  (see :ref:`existing-docker-image`).
```

Comment:
I don't know if 'using existing docker image' is the right place to link to here, since I don't think we have instructions there on adding new things to your image.

I think we should just not link to anything right now, and then clean up our 'building user image' documentation.

Reply:
fixed